nb-apis/adapters-api/src/main/java/io/nosqlbench/adapters/api/activityconfig/yaml/OpData.java

```diff
@@ -29,47 +31,61 @@ public class OpData extends OpTemplate {
     private Map<String, Object> params = Map.of();
     private Map<String, String> bindings = Map.of();
     private Map<String, String> tags = new LinkedHashMap<>();
-    private int refKey;
+    private int refkey;
 
-    public OpData(String desc, String name, Map<String, String> tags, Map<String, String> bindings, Map<String, Object> params, Map<String, Object> op) {
+    public OpData(
+        String desc, String name, Map<String, String> tags, Map<String, String> bindings,
+        Map<String, Object> params, Map<String, Object> op, int refkey
+    ) {
         this.desc = desc;
         this.name = name;
         this.tags = tags;
         this.bindings = bindings;
         this.params = params;
         this.op = op;
+        this.refkey = refkey;
     }
 
-    public OpData() {}
+    public OpData() {
+    }
```

Comment:
Please explain this new `refkey`

Reply:
refkey was a feature added to support efficient caching structures with the multi-client logic before, but it had a fatal error in the first edition, which was that it was being set _statefully_ on something that was lazily derived from skeleton data. (Once off the stackframe, any changes disappeared, thus prepared statements were probably broken except in the case where you had exactly 1 statement under test)

Refkey is intended to be ephemeral to an op template. It is simply a unique numeric identifier which can be used to memoize anything related to the op template in a caching structure. It is specifically an int to allow atomic reference arrays to be used. So, worst case storage is ary[] per space, and there is no hashing needed. Thus, each CQL space has its own sparse array of prepared statement referents, managed atomically and lazily. The same caching logic is used in the space caching implementation as well.

Reply:
The change here was to move the refkey to the skeleton data from the yaml, added immediately by the loader as an enumeration, and exposed in the higher-order op template types as an intrinsic property